elasticsearch/kontena.yml

```diff
@@ -0,0 +1,78 @@
+stack: kontena/elasticsearch
+version: 0.1.0
+description: Elasticsearch
+expose: elastic
+variables:
+  instances:
+    type: integer
+    from:
+      prompt: How many instances?
+  reset_passwords:
+    type: boolean
+    from:
+      prompt: Reset all default passwords during initial stack installation?
+  elastic_password:
+    only_if: reset_passwords
+    type: string
+    from:
+      vault: ELASTIC_PASSWORD
+      random_string: 12
+    to:
+      vault: ELASTIC_PASSWORD
+  kibana_password:
+    only_if: reset_passwords
+    type: string
+    from:
+      vault: KIBANA_PASSWORD
+      random_string: 12
+    to:
+      vault: KIBANA_PASSWORD
+  logstash_password:
+    only_if: reset_passwords
+    type: string
+    from:
+      vault: LOGSTASH_PASSWORD
+      random_string: 12
+    to:
+      vault: LOGSTASH_PASSWORD
+
+services:
+  elastic:
+    image: docker.elastic.co/elasticsearch/elasticsearch:5.5.2
+    instances: {{ instances }}
+    stateful: true
+    mem_limit: 3g
```

Comment:
Should this come from a variable?

Reply:
maybe. by default JVM underneath Elastic hogs 2G so need to also then sync this with JVM mem settings for elastic so that JVM max mem is 0.8 * `mem_limit`. needs some templating magic :)